### PR TITLE
[MHV-56866] Replacing aria-label with aria-described by on links

### PR DESCRIPTION
**Pull request by vsaleem**
- In Secure Messaging, the message details links are not accessible.
- This update replaces the use of aria-label on links to use aria-describedby instead.
- When a screen reader is used on the links, it should read the link's text and the date associated with the message.
- This change is updated on the Inbox, Drafts, Send, and Trash screens.
>
- N/A
>
_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._
>
|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |
>
*Va.gov - Secure Messaging*
>
### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by KyleCardwell — DISMISSED** [review]
LGTM!

File: src/applications/mhv-secure-messaging/components/ThreadList/ThreadListItem.jsx [hunk 1]
```diff
@@ -101,11 +101,7 @@ const ThreadListItem = props => {
       </div>
       <div className="vads-l-col vads-u-margin-left--1">
         <Link
-          aria-label={`${
-            unreadMessages ? 'Unread message. Subject:' : 'Message subject:'
-          } ${categoryLabel}: ${subject}, ${formattedDate()}. ${
-            hasAttachment ? ' Has attachment.' : ''
-          }`}
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/ThreadList/ThreadListItem.jsx:108]: may need to add a child element i.e. `<span class="sr-only">Unread message</span>` to preserve. unread status in aria read out

File: src/applications/mhv-secure-messaging/components/ThreadList/ThreadListItem.jsx [hunk 2]
```diff
@@ -236,11 +233,7 @@ const ThreadListItem = props => {
         </div>
         <Link
           data-dd-action-name="Link to Message Subject Details"
-          aria-label={`${
-            unreadMessages ? 'Unread message.' : ''
-          } Message subject: ${categoryLabel}: ${subject}, ${formattedDate()}. ${
-            hasAttachment ? ' Has attachment.' : ''
-          }`}
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/components/ThreadList/ThreadListItem.jsx:243]: may need to add a child element i.e. `<span class="sr-only">Unread message</span>` to preserve. unread status in aria read out

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
may need to add a child element i.e. `<span class="sr-only">Unread message</span>` to preserve unread status in aria read out in couple places

**Review by powellkerry — DISMISSED** [review]

**Review by alex-morgun-1 — APPROVED** [review]
Looks great 🟢

**Pull request merged**